### PR TITLE
model: Highlight stream name in reporting of moving/splitting topics.

### LINKED ISSUE #1172: Improve reporting upon moving/splitting topics
Currently a message is given to the user when a topic is moved, but we could give more information about this:
- where from?
- where to?
- how many? (all? some?)

For the first version of this, updating the text we output would be sufficient.

As a follow-up, we could consider if there are other ways to indicate eg. split topics.

For comparison, the web app appears to intentionally not sort after splitting topics, to leave both parts in view, whereas ZT does so.

### PR DESCRIPTION
**Pull request by srdeotarse**
**What does this PR do?**  
This PR highlights stream names in reporting of moving/splitting topic in footer text.

Partially fixes #1172 

CZO - https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/.E2.9C.94.20Improve.20reporting.20on.20moving.2Fsplitting.20topics.20.23T1172.20.23T1178

New CZO - https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/Highlight.20text.20of.20reporting.20on.20moving.2Fsplitting.20topics

**Tested?** 
- [x] Manually
- [x] Existing tests (adapted, if necessary)
- [ ] New tests added (for any new behavior)
- [x] Passed linting & tests (each commit)

**Commit flow** 

**Notes & Questions** 

**Interactions** 

**Visual changes** 

Screenshot (66)

### CODE REVIEW

File: zulipterminal/model.py [hunk 1]
```diff
@@ -580,6 +582,30 @@ def update_stream_message(
...
             old_topic = message.get("subject", None)
             new_topic = request["topic"]
+            stream_title_markup = (
+                "footer_contrast",
+                [
+                    (f" {stream_name} {STREAM_TOPIC_SEPARATOR} "),
+                ],
+            )
+            stream_name = urwid.Text(stream_title_markup)
+            stream_name.markup = stream_title_markup
```
Comment (neiljp) [inline zulipterminal/model.py:592]: We can greatly simplify the formatted text here to `(style_string, text)`, which makes this much easier to code - and test.

The same applies to the topic text.

**Review by neiljp — COMMENTED** [review]
@srdeotarse This achieves the formatting fine, but there are many ways to style text in urwid, so passing `["some string", ("named_style", " another string"), ...]` is going to be simpler.

This has been discussed in the past in terms of better defining an actual type for this 'styled text' in other topics and open PRs, which you may find informative - it's rather complex due to what is possible!

One followup aspect to this is that we're casually passing in a list to `report_success` and this is not being flagged by mypy! We may well want to change the type of the function, but I'd like to know why mypy is not warning about this - tests appear to be passing.

File: zulipterminal/model.py [hunk 2]
```diff
@@ -589,9 +615,17 @@ def update_stream_message(
                 else:  # propagate_mode == "change_later":
                     messages_changed = "some messages'"
                 self.controller.report_success(
-                    f"You changed {messages_changed} topic"
-                    + f" from #{stream_name} > {old_topic}"
-                    + f" to #{stream_name} > {new_topic}.",
+                    [
+                        "You changed ",
+                        messages_changed,
+                        " topic from ",
```
[the remark this answers is not in the document]
Comment (srdeotarse) [reply zulipterminal/model.py:621]: @neiljp If f-string is not converted to list of strings, the ``urwid.Text`` is not highlighted and is displayed as ``("style_string", "some_string")``

**Comment by srdeotarse** [comment]
@neiljp ``report_success`` function is passing the string list to ``set_footer_text`` whose argument type is ``Optional[List[Any]]``. Maybe that's why ``mypy`` is not showing type error.
Also, about changing the type of function ``report_success`` to ``Optional[List[Any]]``, is this what you intend?
Because passing ``stream_name`` and ``topic_name`` as a string doesn't display highlighting.

**Comment by srdeotarse** [comment]
@zulipbot add "PR needs review".

**Comment by neiljp** [comment]
> report_success function is passing the string list to set_footer_text whose argument type is Optional[List[Any]]
This doesn't explain why it wasn't showing the error previously. As per my stream comment just now, I suspect this is due to lack of typing between major modules, and we have a few options to what this 'should' be.

File: zulipterminal/model.py (hunk 1 above)
Comment (neiljp) [reply zulipterminal/model.py]: You can see the result of this in the tests, but you've not simplified all the way in the code itself.

File: zulipterminal/model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (neiljp) [reply zulipterminal/model.py:609]: I was only referring to these 3 lines.

**Comment by srdeotarse** [comment]
@zulipbot add "PR needs review".

**Comment by neiljp** [comment]
@srdeotarse The second commit looks fine to merge, but I'm a little concerned the first does not take into account all the call sites?

**Comment by srdeotarse** [comment]
@zulipbot add "PR needs review".

**Comment by srdeotarse** [comment]
@zulipbot add "PR needs review".

**Review by neiljp — COMMENTED** [review]
@srdeotarse There were a few minor points here which I've addressed myself in a quick tidy and pushed back:
- the blacken is in an incorrect commit (but passed overall)
- some strings had been combined (and were overly long)
- in one case the text was already a list, that was passed to report_error

As long as this passes, which I don't expect any problems with, I'll merge shortly after :)

**Comment by neiljp** [comment]
@srdeotarse Thanks for the followup here! :tada:

**Pull request merged**